### ISSUE TITLE
Unsupported timestamp format in downsampling source index

### ISSUE THREAD

**Issue by salvatore-campagna** [elastic/elasticsearch#94085]
### Elasticsearch Version

8.6.1

### Java Version

_bundled_

### OS Version

all

### Problem Description

Trying to downsample an index including a `timestamp` field mapped as `date_nanos` triggers a `MapperParsingException`.

### Steps to Reproduce

We expect to be able to reproduce this issue using a YAML test where we try to downsample an index including a `date_nanos` `timestamp` field.

```
PUT _component_template/my-xroos-testdata-mappings
{
  "template": {
    "mappings": {
      "properties": {
        "correlation_id": {
          "type": "text"
        },
        "anyvalue": {
          "type": "long",
          "time_series_metric": "gauge"
        },
        "source": {
          "type": "keyword",
          "time_series_dimension": true
        },
        "status": {
          "type": "keyword",
          "time_series_dimension": true
        },
        "error_code": {
          "type": "keyword",
          "time_series_dimension": true
        },
        "@timestamp": {
          "type": "date_nanos",
          "format": "uuuu-MM-dd'T'HH:mm:ss.SSSSSSSSSX"
        }
      }
    }
  },
  "_meta": {
    "description": "Mappings for test data"
  }
}

PUT _ilm/policy/my-xroos-testdata-lifecycle-policy
{
  "policy": {
    "phases": {
      "hot": {
        "actions": {
          "rollover": {
            "max_age": "1m",
            "max_primary_shard_size": "5gb"
          }
        }
      },
      "warm": {
        "min_age": "2m",
        "actions": {
          "shrink": {
            "number_of_shards": 1
          },
          "forcemerge": {
            "max_num_segments": 1
          },
          "downsample": {
            "fixed_interval": "1m"
          }
        }
      },
      "delete": {
        "min_age": "30d",
        "actions": {
          "delete": {}
        }
      }
    }
  }
}

PUT _component_template/my-xroos-testdata-settings
{
  "template": {
    "settings": {
      "index.lifecycle.name": "my-xroos-testdata-lifecycle-policy",
      "index.look_ahead_time": "5m",
      "index.codec": "best_compression"
    }
  },
  "_meta": {
    "description": "Index settings for test data"
  }
}

PUT _index_template/my-xroos-testdata-template
{
  "index_patterns": [
    "my-xroos-datatest-*"
  ],
  "data_stream": {},
  "template": {
    "settings": {
      "index.mode": "time_series",
      "index.routing_path": [
        "source",
        "status",
        "error_code"
      ]
    }
  },
  "composed_of": [
    "my-xroos-testdata-mappings",
    "my-xroos-testdata-settings"
  ],
  "priority": 500,
  "_meta": {
    "description": "Template for my test data"
  }
}
DELETE my-xroos-datatest-first
DELETE _data_stream/my-xroos-datatest-first
PUT my-xroos-datatest-first/_bulk
{ "create":{ } }
{ "@timestamp": "2023-02-23T12:47:14.622123123Z", "correlation_id": "wwu3gaac08", "source": "pgw", "status": "success", "anyvalue": 1}
{ "create":{ } }
{ "@timestamp": "2023-02-23T12:47:14.622123124Z", "correlation_id": "reef3aaac08", "source": "pgw", "status": "success", "anyvalue": 2}
```

**Comment by mroos-boku** [comment]
Btw. this isn't just about the special format in the sample. The downsampling also breaks with the built in "strict_date_optional_time_nanos" format.

**Comment by salvatore-campagna** [comment]
A PR (https://github.com/elastic/elasticsearch/pull/94141) has been merged for `8.8.0` to fix the error.
Another PR is open (https://github.com/elastic/elasticsearch/pull/94238) to backport the same fix to `8.7.1`. I am checking with the release manager if we can further backport this to `8.7.0` instead.
